### PR TITLE
i_594 Clear out previous results table so stale results are not shown - fixes #594

### LINKED ISSUE #594: On Judge Test Results, the last tests cases should not be initially displayed
**Is your feature request related to a problem?** 

On view of Test Results previous test cases briefly, until new test cases are presented,

The old testing results can be confuse with new testing results.

**Feature Description**: 

Change to only show new test cases (and to not show previous cases)

**Have you considered other ways to accomplish the same thing?** 

**Do you have any specific suggestions for how your feature would be ***implemented*** in PC^2?**

**Additional context**:

### PR DESCRIPTION
**Pull request by johnbrvc**
### Description of what the PR does
When manually judging a run, after all the runs have completed and pc2 goes to populate the tests results table,
the test results window comes up and contains the previous run's results until the new table model is populated.
This will clear out the table and put a single entry there indicating that "Judging" is still occurring.  

### Issue which the PR fixes
#594 

### Environment in which the PR was developed (OS,IDE, Java version, etc.)
Version 9.9build 20230126 (Thursday, January 26th 2023 16:08 UTC)
Java ver 17.0.5 build 6431
Linux 5.15.0-56-generic (amd64) 

### Precise steps for _testing_ the PR
Find a couple of problems that have a lot of test cases.
Submit runs for both.
Manually judge one of them and wait for the results window to appear.  
Close the judgement windows.
Now judge the other run.
After it finishes running each test case, the test results window will appear.
There should either be single entry that says "Judging" in cyan color or, it will be populated with the correct results,
depending on how fast your PC is.

### CODE REVIEW

**Comment by johnbrvc** [comment]
When reviewing, note the first commit had lots of extra code I had added (duplicated).  I removed most all of it in a subsequent commit to make it simpler.  GIt makes it look like a lot of stuff has been added and removed.  Not really the case.

File: src/edu/csus/ecs/pc2/ui/TestResultsPane.java [hunk 1]
```diff
@@ -1894,6 +1894,44 @@ private void addAnyUnexecutedTestCasesToResultsTable(JTable aResultsTable) {
...
+     * 
+     * @param aResultsTable - a JTable expected to already contain one row for each test case which has been executed
+     */
+    public void resetResultsTable() {
+        if(resultsTable != null) {
+            //get the table model which defines the current table contents
+            TestCaseResultsTableModel tableModel = (TestCaseResultsTableModel) resultsTable.getModel();
+            
+            //get how many total test cases were configured into the problem
+            int totalTestCaseCount = currentProblem.getNumberTestCases();
```
Comment (troy2914) [inline src/edu/csus/ecs/pc2/ui/TestResultsPane.java:1909]: this seems unused?

**Review by troy2914 — CHANGES_REQUESTED** [review]

File: src/edu/csus/ecs/pc2/ui/TestResultsPane.java (hunk 1 above)
Comment (johnbrvc) [reply src/edu/csus/ecs/pc2/ui/TestResultsPane.java]: Yes.  Lazy duplication of code from addAnyUnexecutedTestCases()...  It has been removed and pushed.

**Review by troy2914 — APPROVED** [review]

**Comment by johnbrvc** [comment]
> I executed the "Steps to Test the PR". The changes do indeed appear to suppress bogus data _in the JTable_ until the judging is completed. However, the "Num Failed Test Cases" label in the upper right corner of the TestResultsPane says "All Passed" during the time when the JTable entries contain the cyan "(Judging)" label. See the following screenshot which took a few times to capture (but would occur more commonly on a slower machine). I think that label should either be blank during the "(Judging)" time window, or else should also say "(Judging)"; it's confusing to have it say "All Passed" when in fact the judging hasn't been completed (and may in fact only just have begun).
> 
> I suspect that implementing this would require a change in `TestResultsPane.populateGUI()` around line 1411, adding code like
> 
> ```java
>                 } else  if (judging) {
>                     getNumFailedTestCasesLabel().setForeground(Color.cyan);
>                     getNumFailedTestCasesLabel().setText("(Judging...)");
>                 }
> ```
> 
> although it's not immediately clear to me how `populateGUI()` would have access to the (proposed) `judging` variable which indicates that judging is still under way... But in any case I don't think we should merge this PR without fixing this somehow.
> 
> image

Agree.  Will fix that.

**Review by lane55 — CHANGES_REQUESTED** [review]
Awaiting fix for: Num Failed Test Cases" label

File: src/edu/csus/ecs/pc2/ui/SelectJudgementPaneNew.java [hunk 2]
```diff
@@ -861,7 +861,12 @@ protected void executeRun() {
 
         executable = new Executable(getContest(), getController(), run, runFiles);
 
-        // getManualRunResultsPanel().clear();
+        // only if do not show output is not checked, clear the results pane
+        if (!getContest().getProblem(run.getProblemId()).isHideOutputWindow()) {
```
[the remark this answers is not in the document]
Comment (johnbrvc) [reply src/edu/csus/ecs/pc2/ui/SelectJudgementPaneNew.java:865]: Yes, isHideOutputWindow does precisely that.  On line 906, in fact, that is where it makes the TestResultsFrame visible only if isHideOutputWindow is false (meanin, you want to show the TestResultsFrame).  It does not matter if we clearData anyway, even if we're NOT viewing the TestResultsFrame, but for consistency, I think it makes sense to only clear if we are showing that frame.

File: src/edu/csus/ecs/pc2/ui/TestResultsPane.java [hunk 3]
```diff
@@ -1894,6 +1894,44 @@ private void addAnyUnexecutedTestCasesToResultsTable(JTable aResultsTable) {
...
+            // Add exactly one row as a place holder to tell the user judging is taking place
+            //     "Awaiting Result",                          //result string
+            //      "--  ",                                     //execution time (of which there is none since the test case wasn't executed)
+            //      "",                                         //link to team output (none since it wasn't executed)
+            //      "",                                         //link to team compare-with-judge label (disabled since there's no team output)
+            //      "",                                         //link to judge's output (answer file) if any
+            //      "",                                         //link to judge's data if any
+            //      "",                                         //link to validator stdout (none)
+            //      ""                                          //link to validator stderr (none)
+            TestResultsRowData rowData = new TestResultsRowData("Judging", "--  ","", "","", "", "", "");
```
[the remark this answers is not in the document]
Comment (johnbrvc) [reply src/edu/csus/ecs/pc2/ui/TestResultsPane.java:1924]: Correct.  When new results come in, the table model is reset with the new results data.

**Comment by johnbrvc** [comment]
> I executed the "Steps to Test the PR". The changes do indeed appear to suppress bogus data _in the JTable_ until the judging is completed. However, the "Num Failed Test Cases" label in the upper right corner of the TestResultsPane says "All Passed" during the time when the JTable entries contain the cyan "(Judging)" label. See the following screenshot which took a few times to capture (but would occur more commonly on a slower machine). I think that label should either be blank during the "(Judging)" time window, or else should also say "(Judging)"; it's confusing to have it say "All Passed" when in fact the judging hasn't been completed (and may in fact only just have begun).
> 
> I suspect that implementing this would require a change in `TestResultsPane.populateGUI()` around line 1411, adding code like
> 
> ```java
>                 } else  if (judging) {
>                     getNumFailedTestCasesLabel().setForeground(Color.cyan);
>                     getNumFailedTestCasesLabel().setText("(Judging...)");
>                 }
> ```
> 
> although it's not immediately clear to me how `populateGUI()` would have access to the (proposed) `judging` variable which indicates that judging is still under way... But in any case I don't think we should merge this PR without fixing this somehow.
> 
> image

This has been fixed with the latest commit.

**Review by clevengr — APPROVED** [review]
I re-ran the Steps to Test the PR and verified that, if there is sufficient load on my machine, the "status" label temporarily shows "(Judging)" instead of the status of the previous run. 
PR looks good to me; I approve.

File: src/edu/csus/ecs/pc2/ui/TestResultsPane.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (lane55) [reply src/edu/csus/ecs/pc2/ui/TestResultsPane.java:1924]: thanks.

**Pull request merged**